packages/genesys-spark-components/src/components/beta/gux-avatar-focusable/gux-avatar-focusable.tsx

```diff
@@ -0,0 +1,46 @@
+import { Component, h, JSX, Element } from '@stencil/core';
+import { logWarn } from '@utils/error/log-error';
+
+@Component({
+  styleUrl: 'gux-avatar-focusable.scss',
+  tag: 'gux-avatar-focusable-beta',
+  shadow: true
+})
+export class GuxAvatarLink {
+  @Element() root: HTMLElement;
+
+  componentDidLoad() {
+    this.validateSlot();
+  }
+
+  private validateSlot(): void {
+    const slottedElement = this.root.querySelector('a, button');
+    const avatarElement = slottedElement.querySelector('gux-avatar-beta');
+
+    if (!slottedElement) {
+      logWarn(
+        this.root,
+        'An anchor tag or button tag must be slotted into gux-avatar-focusable'
+      );
+    }
+
+    if (!avatarElement) {
+      logWarn(this.root, 'Slotted element must contain a gux-avatar');
+    }
+
+    if (['small', 'xsmall'].includes(avatarElement.size)) {
+      slottedElement.classList.add('gux-focus-ring-small-focused-visible');
```

Comment:
I think this could be done in a simpler way.

```
  render(): JSX.Element {
    return (
      <div class={`gux-size-${this.avatarElement.size}`}>
        <slot></slot>
      </div>
    ) as JSX.Element;
  }
```
```
.gux-size-small,
.gux-size-xsmall {
  &:focus-visible {
    @include focus.gux-focus-ring-small;
  }
}

....
```